packages/react/src/transactions/useSetProfileMetadata.ts

```diff
@@ -0,0 +1,58 @@
+import { OnchainSetProfileMetadataRequest, Profile } from '@lens-protocol/api-bindings';
+import {
+  PendingSigningRequestError,
+  TransactionKind,
+  UserRejectedError,
+  WalletConnectionError,
+} from '@lens-protocol/domain/entities';
+import { BroadcastingError } from '@lens-protocol/domain/use-cases/transactions';
+import { invariant } from '@lens-protocol/shared-kernel';
+
+import { SessionType, useSession } from '../authentication';
+import { Operation, useOperation } from '../helpers/operations';
+import { AsyncTransactionResult } from './adapters/AsyncTransactionResult';
+import { useSetProfileMetadataController } from './adapters/useSetProfileMetadataController';
+
+export type UseSetProfileMetadataArgs = OnchainSetProfileMetadataRequest;
+
+export type SetProfileMetadataOperation = Operation<
+  AsyncTransactionResult<void>,
+  BroadcastingError | PendingSigningRequestError | UserRejectedError | WalletConnectionError,
+  [UseSetProfileMetadataArgs]
+>;
+
+/**
+ * Set a profile's metadata.
+ *
+ * You MUST be authenticated via {@link useLogin} to use this hook.
+ *
+ * @category Profiles
+ * @group Hooks
+ * @param args - {@link UseSetProfileMetadataArgs}
+ */
+export function useSetProfileMetadata({
```

Comment:
I think we talked about calling it `useUpdateProfileMetadata` to match other similar hooks name
also in other hooks we use useDeferredTask but honestly I don't understand the difference.

from user perspective the difference is in `isPending` vs `loading` returned, so better to have it consistent among similar hooks.

cc @cesarenaldi what shall we use?

Reply:
`useDeferredTask` is a generalization of `useOperation` that can be used also for lazy read hooks.

We should not longer use `useOperation`.